server/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -207,6 +208,8 @@ public class IndicesService extends AbstractLifecycleComponent
         Setting.Property.NodeScope
     );
 
+    static final NodeFeature SUPPORTS_AUTO_PUT = new NodeFeature("indices.supports_auto_put");
```

Comment:
nit: to keep it in line with FeatureInfrastructureFeatures, I propose to change it to "indices.auto_put_supported"

Reply:
Good suggestion